src/libexpr/flake/flakeref.cc

```diff
@@ -102,6 +102,9 @@ std::pair<FlakeRef, std::string> parsePathFlakeRefWithFragment(
 
         if (isFlake) {
 
+            if (!S_ISDIR(lstat(path).st_mode))
+                throw BadURL("path '%s' is not a flake (because it's not a directory)", path);
```

Comment:
We should do at least something nice when user passes a `flake.nix` file. That could be a custom error at minimum, or we could just canonicalize it to its directory.
It seems reasonable to me to assume that a file called `flake.nix` is never a `--file`-like independent expression.

Reply:
I'm all for any kind of extra nicety, but I just want to fix the bug here :)

Reply:
So it turns out that something in the testsuite relied on that. That's a compelling-enough argument to make it more gentle. I've changed that so that pointing to `/flake.nix` just warns